### ISSUE TITLE
to_hierarchical_dataframe() fails on tables that contain indexed columns

### ISSUE THREAD

**Issue by lvsltz** [oruebel/ndx-icephys-meta#53]
Tested on https://github.com/oruebel/ndx-icephys-meta/commit/279bb2e894bc9b10ca9dcb1e6f84a67fc5d6b591

MWE:
```python
ir_index = nwbfile.add_intracellular_recording(electrode=electrode,
                                               stimulus=stimulus,
                                               response=response)

# (B) Add a list of sweeps to the sweeps table
sim_rec_table = nwbfile.get_icephys_simultaneous_recordings()
sim_rec_table.add_column('new_metadata',description = 'metadata as array', index= True)
sweep_index = nwbfile.add_icephys_simultaneous_recording(recordings=[ir_index,], new_metadata = [1,2,3])

nwbfile.icephys_simultaneous_recordings.to_hierarchical_dataframe()
```
because of
```python
...
~/miniconda3/envs/nwb/lib/python3.7/site-packages/ndx_icephys_meta/icephys.py in to_hierarchical_dataframe(self, flat_column_index)
    192 
    193         # Construct the pandas dataframe with the hierarchical multi-index
--> 194         multi_index = pd.MultiIndex.from_tuples(index, names=index_names)
    195         out_df = pd.DataFrame(data=data, index=multi_index, columns=columns)
    196         return out_df

...

TypeError: unhashable type: 'list'
```

**Comment by oruebel** [comment]
Can you test on this branch:

https://github.com/oruebel/ndx-icephys-meta/pull/52

I have not merged that one yet, because I need to add more tests.

**Comment by lvsltz** [comment]
I had checked your changes in the other branch and didn't seem to overlap.

Anyway, I checked, the error is still there:
https://www.dropbox.com/s/dz0kbpitzq06xqp/test_icephys_file.h5?dl=0

**Comment by oruebel** [comment]
Thanks for the script. I was able to reproduce the error. I'll take a look.

**Comment by oruebel** [comment]
@lvsltz this should be fixed now both on the master branch and #52 . I'm planning to merge #52 some time next week, but it would be great if you could start to test with #52.

**Comment by lvsltz** [comment]
Will do.
Thanks for fixing it so quickly.
